R/epidist_utils.R

```diff
@@ -0,0 +1,703 @@
+#' Specify distribution parameter uncertainty
+#'
+#' @description A helper function when creating uncertainty for the parameters
+#' of the distribution for the `epidist` object
+#'
+#' @param ci_limits A numeric vector of length two with the lower and upper bound of
+#' the confidence interval or credible interval
+#' @param ci A numeric specifying the interval for the ci, e.g. 95 is
+#' 95% ci
+#' @param ci_type A character string, either "confidence interval" or "credible
+#' interval"
+#'
+#' @return List of three elements
+#' @export
+#'
+#' @examples
+#' # example with uncertainty for a single parameter
+#' create_epidist_uncertainty(
+#'   ci_limits = c(1, 3),
+#'   ci = 95,
+#'   ci_type = "confidence interval"
+#' )
+#'
+#' # example for multiple parameters
+#' # lengh of list should match number of parameters
+#' list(
+#'   shape = create_epidist_uncertainty(
+#'     ci_limits = c(1,3),
+#'     ci = 95,
+#'     ci_type = "confidence interval"
+#'   ),
+#'   scale = create_epidist_uncertainty(
+#'     ci_limits = c(2, 4),
+#'     ci = 95,
+#'     ci_type = "confidence interval"
+#'   )
+#' )
+#'
+#' # example with unknown uncertainty
+#' # the function can be called without arguments
+#' create_epidist_uncertainty()
+#' # or give NA as the first argument
+#' create_epidist_uncertainty(NA)
+create_epidist_uncertainty <- function(ci_limits = NA_real_, ci, ci_type) {
```

Comment:
I've made an issue (#99) for this topic. I think dedicating a pull request to better handling of uncertainty is required, also to prevent more commits in the current PR.